### ISSUE TITLE
Freelook disabled on Zandorum disables moving mouse in emote wheel vertically

### ISSUE THREAD

**Issue by Zeffies** [buu342/ACS-EmoteSystem#4]
Basically, if you have freelook disabled in the server settings and open the emote wheel, you can only move your mouse side to side meaning you can only really use 3 of the emotes. Moving forward and back with the arrow keys does move the cursor up and down, but this is unintuitive

**Comment by buu342** [comment]
I'm not sure if much can be done in this regard, since I don't _think_ zandro provides another way of getting mouse movement. One alternative would be to bind emote selection to weapon keys or to the scroll wheel.

**Comment by Zeffies** [comment]
oh, okay. thanks for the help!

**Comment by buu342** [comment]
Keeping this open, as a reminder for the future :slightly_smiling_face:

**Comment by souxd** [comment]
Some mods sometimes freezes the player aim but only on the clientside. When the player uses an emote it also unfreezes the player

**Comment by buu342** [comment]
I already do that
https://github.com/buu342/ACS-EmoteSystem/blob/master/ACS/Core/EMOTESYS.acs#L132

**Comment by souxd** [comment]
yes you do, the issue isn't the player being frozen, but when it unfreezes when you use an emote because, for example, there's an aereal invulnerable dash that usually freezes the player aim on the client side so he cant change direction. When you use the emote while on the dash your aim will unfreeze while in the dash. In my opinion that could be fixed on that other (not talking abt the emote wad) wad itself but it would be really cool if the mod worked well out of the box with these kind of wads
VenicesBridge1.2.10/actors/unlimited/YokuMan/YokuAttackBoss.txt
210,1 ``YOKH A 0 SetPlayerProperty(0,1,4)``

load order:
megagame.wad (iwad)
mm8bdm-v6a.pk3
v6a_menudef_3.2-240405-2229a.pk3
classes-v9fh.pk3
gondolapatchv2.6.2h.pk3
gond-fm-60.5.pk3
gond-fm-cloudfix.pk3
VenicesBridge1.2.10.pk3
EmoteSystem-CoreZandronum-v1.wad
EmoteSystem-Bar2oxxEmotes-v2b.wad

playerclass yokuman;spectate;join

https://github.com/user-attachments/assets/2932e8e4-17bb-4ef9-a6c1-50cd9458ac99

**Comment by souxd** [comment]
I think a check to find out if the player camera is already frozen so you dont need to unfreeze and a check to freeze (and to allow the wheel to unfreeze) the player if he's unfrozen while on the wheel menu could work, cant think of a better solution atm

**Comment by buu342** [comment]
Yeah this is a tough fix, but I think it's possible. I'd basically have to check the player's frozen state serverside and then network that data to the client rather than making assumptions about the player's state. It's never going to be 100% because of the large disconnect between the client and server in Zandronum, but c'est la vie.

**Comment by souxd** [comment]
it only needs checks for when to unfreeze so i dont think it will be that bad of a fix. I don't have free time right now but i believe all of that could be avoided by not freezing the player in the first place which might look weird for the player but eh, he's the one that decided to press the emotes button so thats what he will be working with
